src/edu/washington/escience/myria/operator/network/distribute/IdentityPartitionFunction.java

```diff
@@ -0,0 +1,42 @@
+package edu.washington.escience.myria.operator.network.distribute;
+
+import java.util.BitSet;
+
+import javax.annotation.Nonnull;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import edu.washington.escience.myria.storage.TupleBatch;
+
+/**
+ * Implementation that uses multiple fields as the key to hash The partition of
+ * a tuple is decided by the hash code of a group of fields of the tuple.
+ */
+public final class IdentityPartitionFunction extends PartitionFunction {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /** the column index for partitioning. */
+  @JsonProperty private final int index;
+
+  /**
+   * @param index the column index for partitioning.
+   */
+  public IdentityPartitionFunction(final int index) {
+    this.index = index;
+  }
+
+  @Override
+  public TupleBatch[] partition(@Nonnull final TupleBatch tb) {
+    BitSet[] partitions = new BitSet[numPartition()];
+    for (int i = 0; i < tb.numTuples(); i++) {
+      partitions[tb.getInt(index, i) - 1].set(i);
```

Comment:
We don't have the schema until this method is called, so I don't think we can do it early unless we do it in the Producer: when its schema is set, we check if its distribute function is an 'IdentityDistributeFunction`. An out-of-bound value in the "partition" column will trigger an IndexOutOfBoundException for the array `partitions`, but not making BitSets grow since their sizes are set by `numTuples()` not `numPartitions()`.